### ISSUE TITLE
poetry version solving failed on poetryup and typer

### ISSUE THREAD

**Issue by Tobi-De** [MousaZeidBaker/poetryup#73]
Hi, sorry to bother you and frankly I'm sure it's poetry's fault and not `poetryup`.
Here is the message I get when I try to install the latest version of typer (0.7.0) when poetryup is already installed.

```shell
Because poetryup (0.12.5) depends on typer (>=0.4.1,<0.5.0)
 and no versions of poetryup match >0.12.5,<0.13.0, poetryup (>=0.12.5,<0.13.0) requires typer (>=0.4.1,<0.5.0).
So, because max depends on both typer (^0.7.0) and poetryup (^0.12.5), version solving failed.
```

This makes no sense to me.

**Comment by MousaZeidBaker** [comment]
Thanks for reporting an issue @Tobi-De 

The problem is that `poetryup` itself depends on `typer <0.5.0` but you are trying to install `typer >0.5.0` in the same environment of `poetryup` so you end up with a conflict. 

One solution would be to install `poetryup` in a separate environment, for instance globally in your system using `pip`. However, 
`poetryup` have been migrated to [poetry-plugin-up](https://github.com/MousaZeidBaker/poetry-plugin-up) and I strongly recommend using that instead. Instead of being a standalone tool, it can now be used as a plugin under Poetry. Just make sure to install Poetry the recommended way (with curl) in order to avoid this kind of conflicts, see poetry installation docs for more details: https://python-poetry.org/docs/#installation

**Comment by MousaZeidBaker** [comment]
From [poetry docs](https://python-poetry.org/docs/dependency-specification/#caret-requirements)

> Caret requirements allow [SemVer](https://semver.org/) compatible updates to a specified version. An update is allowed if the new version number does not modify the **left-most non-zero** digit in the major, minor, patch grouping. For instance, if we previously ran poetry add requests@^2.13.0 and wanted to update the library and ran poetry update requests, poetry would update us to version 2.14.0 if it was available, but would not update us to 3.0.0. If instead we had specified the version string as ^0.1.13, poetry would update to 0.1.14 but not 0.2.0. 0.0.x is not considered compatible with any other version.
